config/metadata/src/main/java/io/helidon/config/metadata/Configured.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (c) 2021 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.helidon.config.metadata;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Inherited;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * A configured class can load its values from configuration.
+ * There are two options for a supported type - either it is an interface/class
+ * with a {@code public static X create(Config)}, or a builder class with
+ * a method {@code public Builder config(Config)}.
+ *
+ * This annotation is used to provide IDE autocompletion, and to build documentation metadata
+ * using an annotation processor.
+ */
+@Target(ElementType.TYPE)
+@Inherited
+@Retention(RetentionPolicy.RUNTIME)
```

Comment:
The annotations are not examined at runtime, are they? Why `RUNTIME` retention?

Reply:
Using `SOURCE` as that is sufficient for annotation processing.